pkgs/development/python-modules/numcodecs/default.nix

```diff
@@ -54,7 +54,7 @@ buildPythonPackage rec {
     msgpack
   ];
 
-  pytestFlagsArray = [ "$out/${python.sitePackages}/numcodecs" ];
+  preCheck = "cd $out";
```

Comment:
I don't like the idea of `cd` to a directory when we could do it with flags. In my experience, it is pretty stateful and is a source of surprises. Even if `cd` is the most feasible implementation, we should `cd` back during `postCheck`.

Reply:
Then, I agree that `cd` would be a short-term solution to push this work forward.

Still, for packages that didn't have `cd` in `preCheck` before this change, it seems better to use `pushd`, so that we could `popd` afterward in `postCheck`.